### ISSUE TITLE
Nothing asserts an intervention left the non-last positions alone, so a whole-prompt intervention would pass

### ISSUE THREAD

**Issue by speediedan** [speediedan/interpretune#441]
Surfaced while measuring cross-backend intervention parity for the interp-engine adapter component, and the same defect class we have been tracking all
week — this time arriving through a **default** rather than a name.

## What the parity measurement found

interp-engine steers **every prompt position** by default. Interpretune's intervention contract is the
**last token**. The two agreed about the arithmetic of each operation while computing *different operations*.
Nothing raised: shapes matched, the forward completed, only the meaning differed. Measured on gpt2 layer 5 —
without a position mask, 5 of 5 positions change; with one, exactly 1.

A hook-name mismatch at least gives you two names to compare. **A default gives you nothing to look at.**

## The gap on our side

`apply_intervention_to_last_token` is correct by construction — it slices `value[:, last_pos, ...]`. But it
is called **per backend, by convention**:

```
adapters/nnsight/backends.py           6 call sites
adapters/transformer_lens/backends.py  1 call site
```

Nothing requires a backend to call it, and **nothing verifies that one did**. The closest existing
assertion is `tests/core/test_analysis_backends.py:1389`:

```python
assert torch.allclose(activation[0, 2, :], replacement_vec), "Replace mode should overwrite last position"
```

That asserts the last position **was** changed. It does not assert the others were **not**. A backend
applying the intervention to every position passes it — the last position is still correct, and positions 0
and 1 are never inspected.

**A whole-prompt intervention produces entirely plausible logits**, so a value comparison alone cannot catch
this. Verified there is no position-count assertion anywhere in the suite (grep control: the same query finds
3/2/7 occurrences of the helper across three test files, so the search surface is live).

## Proposed

A cross-backend contract assertion on the **count**, not only the values:

- with an intervention applied, **exactly one** position differs from an un-intervened reference;
- and it is `last_pos`.

The adapter component added exactly this and **verified it fires** by removing their mask — two tests
fail, including the count rather than only the values. The count was asserted deliberately, so the guard's passing means something.

This is worth having as a shared contract test rather than per backend, because the failure it catches is a
backend *not* participating — which a per-backend test written by that backend's author is least likely to
cover.

## Scope note

No evidence any bundled backend currently gets this wrong; all call sites look correct. This is about the
absence of a guard, not a known break. The urgency comes from hub-delivered backends, where the author is
not in this repo and the contract is currently communicated only by a helper's name.
